pennylane/ops/op_math/condition.py

```diff
@@ -666,17 +667,39 @@ def _get_cond_qfunc_prim():
     cond_prim.multiple_results = True
 
     @cond_prim.def_impl
-    def _(conditions, *args_and_consts, jaxpr_branches, n_consts_per_branch, n_args):
-
-        args = args_and_consts[:n_args]
-        consts_flat = args_and_consts[n_args:]
+    def _(*all_args, jaxpr_branches, n_consts_per_branch, n_args):
+        n_branches = len(jaxpr_branches)
+        conditions = all_args[:n_branches]
+        args = all_args[n_branches : n_branches + n_args]
+        consts_flat = all_args[n_branches + n_args :]
+
+        if isinstance(conditions[0], qml.measurements.MeasurementValue):
+            if len(conditions) > 2:
+                raise ConditionalTransformError(
+                    "'elif' branches are not supported when not using @qjit and the "
+                    "conditional includes mid-circuit measurements."
+                )
+            conditions = (conditions[0], ~conditions[0])

```

Comment:
Doesn't have to be this PR, but could elifs be supported with additional quantum `Conditional` operators? I'm thinking:
```
if m0:
   qml.X(0)
elif m1:
  qml.Y(0)
else:
  qml.Z(0)
```
would be equivalent to:
```
Conditional(m0, qml.X(0))
Conditional(not m0 and m1, qml.Y(0), qml.Z(0))
```
which should be extendible to any number of elifs.

Reply:
Should be doable in this PR, I'll try it out and re-request review.

Reply:
@dime10 Added